### PR TITLE
Validation

### PR DESCRIPTION
**Pull request by OLIPLICHE**
- I add validations to my models
- I created specs for them and find and fix N+1 queries problems.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -3,6 +3,11 @@ class Post < ApplicationRecord
   has_many :comments, foreign_key: 'post_id'
   has_many :likes, foreign_key: 'post_id'
 
+  validates :Title, presence: true
```
Comment (petudeveloper) [inline app/models/post.rb]: Please add the proper validation in order for your **title** to not exceed 250 characters. Remember that validation provides accuracy, details, and clarity to your data.

**Review by petudeveloper — CHANGES_REQUESTED** [review]
Hi @OLIPLICHE , 

Good job so far! Your `User` model has validations, your `Post` model has validations, you have unit tests for `User`, `Post`, `Comment` and `Like` model.

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [ ] _Check the comments under the review._

- [x] Your README file should be as descriptive as possible, please try adding more details about how to use your app. For example please try adding information on how to install, how to use it, how to run tests, and how to run a local development server. This is a good practice because it helps the more inexpert developers to use your app.
**_Note_** This comment is embedded here because I am not able to add this comment to the corresponding file.

- [x] When I try to migrate the database in order to use your app an error displays. Check the image below to have a better understanding:
Capture
**_Note_** This comment is embedded here because I am not able to add this comment to the corresponding file.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/models/post.rb (hunk 1 above)
Comment (OLIPLICHE) [reply app/models/post.rb]: Thank you. the issue has been fixed

Comment (OLIPLICHE) [reply app/models/post.rb]: review


Areview

Comment (OLIPLICHE) [reply app/models/post.rb]: Migration works well at my end and the server is running actually

**Review by mke2111 — CHANGES_REQUESTED** [review]
### Hey @OLIPLICHE, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️

- [ ] Unfortunately, I'm still failing to migrate your files just like the previous reviewer. Please look into it.
Screenshot from 2021-12-19 01-27-13


_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # blog-app
-The Blog app is a classic example of a blog website. I create a fully functional website that shows the list of posts and empower readers to interact with them by adding comments and liking posts.
+The Blog app is a classic example of a blog website. It's a functional website that shows the list of posts and empower readers to interact with them by adding comments and liking posts.
 
 ## Getting Started
  Go ahead and check the most recent version of each of this in the code editor:
```
Comment (mke2111) [inline README.md]: - [x] In the getting started section, add the complete instructions on how to set up locally. Remember to include all the necessary commands.

Comment (OLIPLICHE) [reply README.md]: God

Comment (OLIPLICHE) [reply README.md]: Migration work well at my end. the code is working.

Comment (OLIPLICHE) [reply README.md]: ta

Comment (OLIPLICHE) [reply README.md]: tc

Comment (mke2111) [reply README.md]: Let me look further into this and also set up your repo on another computer. I'll get back to you in a few.

Comment (OLIPLICHE) [reply README.md]: okay sir. Thank you

Comment (mke2111) [reply README.md]: You have this migration file `rename_column :posts, :string, :title` and apparently there is no table column of string.
Remember the format shouls be like `rename_column :table, :old_column, :new_column`. For your case, the old column name string does not exist. Check the migration file `FixTitleInPost`

This is the reason for this error.

**Comment by Davidosky007** [comment]
## Hi @OLIPLICHE 👋, 

## Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 

thank-you



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 2 above)
Comment (OLIPLICHE) [reply README.md]: That was exactly the issue. It has been solved. thank you. @mke2111

**Pull request merged**